orderbook/src/api/post_quote.rs

```diff
@@ -1,19 +1,96 @@
-use crate::api;
+use crate::{
+    api::{
+        self,
+        validation::{PreOrderData, PreOrderValidator, ValidationError},
+    },
+    fee::MinFeeCalculating,
+};
 use anyhow::{anyhow, Result};
+use chrono::{DateTime, Utc};
 use ethcontract::{H160, U256};
 use model::{
     app_id::AppId,
     order::{BuyTokenDestination, OrderKind, SellTokenSource},
     u256_decimal,
 };
 use serde::{Deserialize, Serialize};
+use shared::price_estimation::{self, PriceEstimating, PriceEstimationError};
 use std::convert::Infallible;
+use std::sync::Arc;
+use warp::reply::Json;
 use warp::{hyper::StatusCode, reply, Filter, Rejection, Reply};
 
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Fee {
+    #[serde(with = "u256_decimal")]
+    pub amount: U256,
+    pub expiration_date: DateTime<Utc>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FeeResponse {
+    fee: Fee,
+    side: OrderQuoteSide,
+}
+
+#[derive(Debug)]
+pub enum FeeError {
```

Comment:
Ubernit: Maybe this should be a `quote` error as its an error from the new Quote API?